app/admin/views.py

```diff
@@ -152,18 +152,32 @@ def user_list():
 @login_required
 @roles_required('admin')
 def user_list_csv():
-    drivers = Person.query.filter(Carpool.driver_id == Person.id)
-    passengers = Person.query.filter(RideRequest.status == 'approved').\
-        filter(RideRequest.person_id == Person.id)
-    output = io.StringIO()
+    output = io.BytesIO()
     writer = csv.writer(output)
     writer.writerow(['Nomad carpool drivers and riders'])
-    writer.writerow(['name', 'email', 'phone', 'preferred'])
-    for user in drivers.union(passengers).order_by(Person.name):
-        writer.writerow([user.name, user.email, user.phone_number,
-            user.preferred_contact_method])
+    writer.writerow(['destination', 'carpool leave time', 'carpool return time',
+                     'name', 'email', 'phone', 'preferred contact method'])
+    query = '''
```

Comment:
Just curious--why write this query as a raw string instead of sqlalchemy?

Reply:
I'm sympathetic to how difficult it is to figure out the SQLAlchemy incantations to build queries like this. If someone wants to take on rebuilding this in SQLAlchemy I'm all for it, but I'm going to merge it as-is.

Reply:
Mainly because I don't know how to write that in SQLAlchemy: it gets data from four tables (destinations, carpools, people, riders). Even if I could, I doubt it would be easier to read, which for me is one of the main reasons to use SQLAlchemy. I could break it up, but running queries in loops (for example, get riders for each carpool though the `riders` relation) quickly leads to performance issues.